pkg/koordlet/util/meminfo.go

```diff
@@ -148,6 +162,60 @@ func readMemInfo(path string, isNUMA bool) (*MemInfo, error) {
 	return &info, nil
 }
 
+func GetHugePagesInfo(nodeDir string) (map[uint64]*HugePagesInfo, error) {
+	hugePagesInfo := map[uint64]*HugePagesInfo{
+		Hugepage1Gkbyte: {
+			NumPages: 0,
+			PageSize: Hugepage1Gkbyte,
+		},
+		Hugepage2Mkbyte: {
+			NumPages: 0,
+			PageSize: Hugepage2Mkbyte,
+		},
+	}
+	hugepageDir := system.GetNUMAHugepagesDir(nodeDir)
+	hugeDirs, err := os.ReadDir(hugepageDir)
+	if err != nil {
+		klog.Warningf("failed to read hugepage dir, err: %w", err)
+		return hugePagesInfo, nil
+	}
+
+	for _, st := range hugeDirs {
+		nameArray := strings.Split(st.Name(), "-")
+		pageSizeArray := strings.Split(nameArray[1], "kB")
+		pageSize, err := strconv.ParseUint(pageSizeArray[0], 10, 64)
+		if err != nil {
+			klog.Warningf("hugepage pageSize parse failed, %s", st.Name())
```

Comment:
need to log the full information including dir path and parsed string, or throw the error